pkg/collector/worker/worker.go

```diff
@@ -169,9 +169,13 @@ func (w *Worker) Run() {
 
 		if sender != nil && !longRunning && config.Datadog.GetBool("integration_check_status_enabled") {
 			sender.ServiceCheck(serviceCheckStatusKey, serviceCheckStatus, hname, serviceCheckTags, "")
-			sender.Commit()
 		}
 
+		// FIXME(remy): this `Commit()` should be part of the above `if`, we keep
+		// it here for now to make sure it's not breaking any historical behavior
+		// with the shared default sender.
+		sender.Commit()
```

Comment:
@remeh It seems that the sender can be nil here.

Reply:
Based on above code https://github.com/DataDog/datadog-agent/pull/18148/files#diff-fa837122b5056960e5f1bcbe5e6da5ed11701b8e75a2dff72681905d9a1e0314R150-R153

when sender cannot be retrieve, it only logs an error, but continue the code flow.

Reply:
Oops good catch. Thank you, sending a fix.

Reply:
@remeh Also, should we keep `!longRunning` condition too, I'm not sure what's the impact of running `sender.Commit()` for longRunning checks 🤔